plutus-core/plutus-ir/Language/PlutusIR.hs

```diff
@@ -66,11 +64,13 @@ data Datatype tyname name uni fun a = Datatype a (TyVarDecl tyname a) [TyVarDecl
     deriving (Functor, Show, Generic)
 
 instance ( PLC.Closed uni
-         , uni `PLC.Everywhere` Serialise
-         , Serialise a
-         , Serialise tyname
-         , Serialise name
-         ) => Serialise (Datatype tyname name uni fun a)
+         , uni `PLC.Everywhere` Flat
+         , Flat a
+         , Flat tyname
+         , Flat name
+         -- This was needed only for the Flat instance
+         , Flat fun
+         ) => Flat (Datatype tyname name uni fun a)
```

Comment:
How come we now need `Flat fun`? `Datatype` is two `TyVarDecl`s (each doesn't reference `fun`) and a `VarDecl` (doesn't use the `fun` that it's parameterized by).

We should probably drop `fun` from `VarDecl`, I introduced it by mistake. I'll do that once this PR is merged. Pinging @michaelpj

Reply:
Removing it from `VarDecl` seems fine. I vaguely remember we included it in some places we didn't need it because it helped type inference? Or am I makin that up?

Reply:
I'd included it in one place where it wasn't needed, but then we talked and I removed it. Not sure about `VarDecl`, this looks like an accidental change (I did `s/name name uni ann/name name uni fun ann`, i.e. whereever there's term-level names and universes, there's most likely built-in functions).